File: apina-core/src/test/java/fi/evident/apina/utils/PropertyUtilsTest.java
package fi.evident.apina.utils;

import org.junit.Test;

import static fi.evident.apina.utils.PropertyUtils.propertyNameForGetter;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class PropertyUtilsTest {

    @Test
    public void propertyNamesForGetters() {
        assertThat(propertyNameForGetter("getFoo"), is("foo"));
        assertThat(propertyNameForGetter("getFooBar"), is("fooBar"));
        assertThat(propertyNameForGetter("isFoo"), is("foo"));
        assertThat(propertyNameForGetter("isFooBar"), is("fooBar"));
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/utils/PropertyUtilsTest.kt
package fi.evident.apina.utils

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class PropertyUtilsTest {

    @Test
    fun `property names for value getters`() {
        assertEquals("foo", propertyNameForGetter("getFoo-11MJ8YI"))
        assertEquals("fooBar", propertyNameForGetter("getFooBar-11MJ8YI"))
        assertEquals("foo", propertyNameForGetter("isFoo-11MJ8YI"))
        assertEquals("fooBar", propertyNameForGetter("isFooBar-11MJ8YI"))
    }

    @Test
    fun `property names for getters`() {
        assertEquals("foo", propertyNameForGetter("getFoo"))
        assertEquals("fooBar", propertyNameForGetter("getFooBar"))
        assertEquals("foo", propertyNameForGetter("isFoo"))
        assertEquals("fooBar", propertyNameForGetter("isFooBar"))
    }
}
